### ISSUE TITLE
gatsby-transformer-json: If objects have `id`s and values of `String(id)` are same, they are recognized as a same object

### ISSUE THREAD

**Issue by gh640** [gatsbyjs/gatsby#33167]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Summary

I used `gatsby-transformer-json` to load a JSON file that has the content like the following:

```json
[
  { "id": { ... },  "name": "a" },
  { "id": { ... },  "name": "b" },
  { "id": { ... },  "name": "c" }
]
```

I tried to fetch all the items with GraphQL but got only the first item. I figured out why it occurs only after seeing the package's code https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-transformer-json/src/gatsby-node.js .

I'd be nice if this behavior is explained in the package's README.

### Steps to Resolve this Issue

1. Improve the package's README file.

**Comment by LekoArts** [comment]
I'm closing this in favor of https://github.com/gatsbyjs/gatsby/issues/28846 as https://github.com/gatsbyjs/gatsby/pull/28942 should fix this
